yt_dlp/extractor/tencent.py

```diff
@@ -97,21 +130,36 @@ def _extract_video_native_subtitles(self, api_response, subtitles_format):
 
         return subtitles
 
-    def _extract_all_video_formats_and_subtitles(self, url, video_id, series_id):
-        formats, subtitles = [], {}
-        for video_format, subtitle_format, video_quality in (
-                # '': 480p, 'shd': 720p, 'fhd': 1080p
-                ('mp4', 'srt', ''), ('hls', 'vtt', 'shd'), ('hls', 'vtt', 'fhd')):
-            api_response = self._get_video_api_response(
-                url, video_id, series_id, subtitle_format, video_format, video_quality)
+    def _extract_all_video_quality(self, api_response):
+        quality = []
+        formats_response = api_response['fl']['fi']
+        for f in formats_response:
+            quality.append(f.get('name'))
+        return quality
```

Comment:
It's really needed this function? You only use it once

maybe it can be:
```python
qualities = [format['name'] for format in api_response['fl']['fi']]
```

Reply:
```py
qualities = traverse_obj(api_response, ('fl', 'fi', ..., 'name'))
```
?

Reply:
Sure, qualities should be fatal but we can add some fallbacks

```python
qualities = traverse_obj(api_response, ('fl', 'fi', ..., 'name')) or ('sd', 'shd', 'fhd')
```

Reply:
I'm not a Python programmer, so I'm not familiar with some of Python's syntax, but I'll modify it using your code.